### PR TITLE
license: refactor license check to be agnostic of input

### PR DESCRIPTION
**Pull request by msanft**
### Context

We want to refactor the apply command to strictly separate the logic working with user interaction (to the `cli` package) and the "business logic", which is agnostic of wether it's called from the CLI (to the `internal/constellation` package) in preparation for the Terraform provider.

### Proposed change(s)

- Refactor the license check to not consume a file handler by default and only check the license in it, regardless of where it comes from.
- Move all user-I/O code to a separate function in the `constellation apply` command in the CLI.

### Additional info

- [AB#3557](https://dev.azure.com/Edgeless/ae37573d-ccde-4af2-ab1e-001e587197d1/_workitems/edit/3557)
- [Working Telemetry](https://teams.microsoft.com/l/message/19:0275989e375a46ceb186e68dce850e9f@thread.skype/1701262125530?tenantId=adb650a8-5da3-4b15-b4b0-3daf65ff7626&groupId=62348c88-6944-413d-9232-9d3adb03009e&parentMessageId=1701262125530&teamName=Edgeless&channelName=Constellation%20Telemetry&createdTime=1701262125530)
- [E2E Test](https://github.com/edgelesssys/constellation/actions/runs/7032780048)

### Checklist

- [x] Add labels (e.g., for changelog category)
- [x] Is PR title adequate for changelog?
- [x] Link to Milestone

### CODE REVIEW

**Review by malt3 — APPROVED** [review]

File: cli/internal/cmd/apply.go [hunk 1]
```diff
@@ -429,6 +433,43 @@ func (a *applyCmd) apply(
 	return nil
 }
 
+// checkLicenseFile reads the local license file and checks it's quota
+// with the license server. If no license file is present or if errors
+// occur during the check, the user is informed and the community license
+// is used.
+func (a *applyCmd) checkLicenseFile(cmd *cobra.Command, csp cloudprovider.Provider) {
```
Comment (thomasten) [inline cli/internal/cmd/apply.go:440]: this must be excluded from the oss build (like before)

**Comment by msanft** [comment]
[New E2E Test](https://github.com/edgelesssys/constellation/actions/runs/7035121155)

**Review by daniel-weisse — APPROVED** [review]

File: internal/license/file_test.go [hunk 2]
```diff
@@ -9,72 +9,49 @@ package license
...
 				require.NoError(err)
 			}
-
-			license, err := FromFile(testFS, tc.licenseFilePath)
-			if tc.wantError {
-				assert.Error(err)
-				return
-			}
-			assert.NoError(err)
-			assert.Equal(tc.wantLicense, license)
```
Comment (thomasten) [inline internal/license/file_test.go:77]: Please readd testing of the functionality by checking the return value. This is more important than covering every error case.

**Review by thomasten — APPROVED** [review]
Thanks :)

**Pull request merged**